In [265]:
import pandas as pd

In [266]:
#df= pd.read_table('C:/Users/sumit/Desktop/ISB/Classwork/Python/Assignment/restaurant.tsv' )
#print(df)

#with open('C:/Users/sumit/Desktop/ISB/Classwork/Python/Assignment/restaurant.tsv') as file:
#    tsv_file = csv.reader(file, delimiter="\t")
#    for line in tsv_file:
#       print(line)
        
df = pd.read_csv('C:/Users/sumit/Desktop/ISB/Classwork/Python/Assignment/restaurant.tsv', sep='\t')
df

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

In [267]:
df.head()
df.tail(200)

order_id  quantity                     item_name  \
4422      1763         1             Canned Soft Drink   
4423      1764         2                  Chicken Bowl   
4424      1764         1           Chips and Guacamole   
4425      1764         1  Chips and Fresh Tomato Salsa   
4426      1764         1                         Chips   
...        ...       ...                           ...   
4617      1833         1                 Steak Burrito   
4618      1833         1                 Steak Burrito   
4619      1834         1            Chicken Salad Bowl   
4620      1834         1            Chicken Salad Bowl   
4621      1834         1            Chicken Salad Bowl   

                                     choice_description item_price  
4422                                             [Coke]     $1.25   
4423  [Fresh Tomato Salsa, [Rice, Fajita Vegetables,...    $22.50   
4424                                                NaN     $4.45   
4425                                                NaN     $2.95   
4426                                                NaN     $2.15   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[200 rows x 5 columns]

In [268]:
df.describe()

order_id     quantity
count  4622.000000  4622.000000
mean    927.254868     1.075725
std     528.890796     0.410186
min       1.000000     1.000000
25%     477.250000     1.000000
50%     926.000000     1.000000
75%    1393.000000     1.000000
max    1834.000000    15.000000

In [269]:
df.shape

(4622, 5)

In [270]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


In [271]:
df['item_name'].unique()

array(['Chips and Fresh Tomato Salsa', 'Izze', 'Nantucket Nectar',
       'Chips and Tomatillo-Green Chili Salsa', 'Chicken Bowl',
       'Side of Chips', 'Steak Burrito', 'Steak Soft Tacos',
       'Chips and Guacamole', 'Chicken Crispy Tacos',
       'Chicken Soft Tacos', 'Chicken Burrito', 'Canned Soda',
       'Barbacoa Burrito', 'Carnitas Burrito', 'Carnitas Bowl',
       'Bottled Water', 'Chips and Tomatillo Green Chili Salsa',
       'Barbacoa Bowl', 'Chips', 'Chicken Salad Bowl', 'Steak Bowl',
       'Barbacoa Soft Tacos', 'Veggie Burrito', 'Veggie Bowl',
       'Steak Crispy Tacos', 'Chips and Tomatillo Red Chili Salsa',
       'Barbacoa Crispy Tacos', 'Veggie Salad Bowl',
       'Chips and Roasted Chili-Corn Salsa',
       'Chips and Roasted Chili Corn Salsa', 'Carnitas Soft Tacos',
       'Chicken Salad', 'Canned Soft Drink', 'Steak Salad Bowl',
       '6 Pack Soft Drink', 'Chips and Tomatillo-Red Chili Salsa', 'Bowl',
       'Burrito', 'Crispy Tacos', 'Carnitas Crispy Tacos

In [272]:
df.isnull().sum() ### This gives max null values in the Choice_description column 

order_id                 0
quantity                 0
item_name                0
choice_description    1246
item_price               0
dtype: int64

In [273]:
df.describe(include='object')

item_name choice_description item_price
count           4622               3376       4622
unique            50               1043         78
top     Chicken Bowl        [Diet Coke]     $8.75 
freq             726                134        730

In [274]:
df['item_price'].describe()   ### Max Order Value in the Restaurant is $8.75 per Unit 

count       4622
unique        78
top       $8.75 
freq         730
Name: item_price, dtype: object

df['item_name'].describe() ## Chicken Bowl is the max ordered from dataset and 726 is the quantity 

In [276]:
#d.	What were the sales of the restaurant during the entire period of the dataset? 

#replaceing the dollar sign

df['item_price'] = df['item_price'].str.replace('$', '')

In [277]:
df["item_price"] = pd.to_numeric(df['item_price'])

In [278]:
df['item_price']

0        2.39
1        3.39
2        3.39
3        2.39
4       16.98
        ...  
4617    11.75
4618    11.75
4619    11.25
4620     8.75
4621     8.75
Name: item_price, Length: 4622, dtype: float64

In [279]:

df["Sales"] = df["quantity"] * df["item_price"]

df.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description  item_price  Sales  
0                                                NaN        2.39   2.39  
1                                       [Clementine]        3.39   3.39  
2                                            [Apple]        3.39   3.39  
3                                                NaN        2.39   2.39  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98  33.96

In [280]:
sum(df["Sales"])#Total sales of the restaurant during the entire period of the dataset 

39237.020000000055

In [281]:
#e.	On average, how much does the restaurant make on any order?
df_Avg = df.groupby('order_id').sum()
df_Avg

quantity                                          item_name  \
order_id                                                                
1                4  Chips and Fresh Tomato SalsaIzzeNantucket Nect...   
2                2                                       Chicken Bowl   
3                2                          Chicken BowlSide of Chips   
4                2                      Steak BurritoSteak Soft Tacos   
5                2                   Steak BurritoChips and Guacamole   
...            ...                                                ...   
1830             2                        Steak BurritoVeggie Burrito   
1831             3                    Carnitas BowlChipsBottled Water   
1832             2              Chicken Soft TacosChips and Guacamole   
1833             2                         Steak BurritoSteak Burrito   
1834             3  Chicken Salad BowlChicken Salad BowlChicken Sa...   

                                         choice_description  item_price  Sales  
order_id                                                                        
1                                       [Clementine][Apple]       11.56  11.56  
2         [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98  33.96  
3         [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...       12.67  12.67  
4         [Tomatillo Red Chili Salsa, [Fajita Vegetables...       21.00  21.00  
5         [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...       13.70  13.70  
...                                                     ...         ...    ...  
1830      [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...       23.00  23.00  
1831      [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       12.90  12.90  
1832       [Fresh Tomato Salsa, [Rice, Cheese, Sour Cream]]       13.20  13.20  
1833      [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...       23.50  23.50  
1834      [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...       28.75  28.75  

[1834 rows x 5 columns]

In [282]:
df_Avg.Sales.mean()

21.39423118865867